src/v2i-hub/SpatPlugin/test/TestSignalControllerConnection.cpp

```diff
@@ -0,0 +1,613 @@
+                            {"SignalGroupId":4,"Phase":4,"Type":"vehicle"},
+                            {"SignalGroupId":5,"Phase":5,"Type":"vehicle"},
+                            {"SignalGroupId":6,"Phase":6,"Type":"vehicle"},
+                            {"SignalGroupId":7,"Phase":7,"Type":"vehicle"},
+                            {"SignalGroupId":8,"Phase":8,"Type":"vehicle"},
+                            {"SignalGroupId":9,"Phase":2,"Type":"pedestrian"},
+                            {"SignalGroupId":10,"Phase":4,"Type":"pedestrian"},
+                            {"SignalGroupId":11,"Phase":6,"Type":"pedestrian"},
+                            {"SignalGroupId":12,"Phase":8,"Type":"pedestrian"}
+                        ]
+                    }
+                )";
+                signalControllerConnection = std::make_unique<SignalControllerConnection>("127.0.0.1", 5000, signalGroupMapping, "", 5020, "someIntersection", 9001);
+                mockSnmpClient = std::make_shared<unit_test::mock_snmp_client>("127.0.0.1", 6045, "administrator", "", "", "");
+                mockUdpServer = std::make_shared<tmx::utils::MockUpdServer>();
+                signalControllerConnection->scSNMPClient = mockSnmpClient;
+                signalControllerConnection->spatPacketReceiver = mockUdpServer;
+            }
+
+            std::vector<char> read_binary_file(std::string name) 
+            {
+                std::ifstream file(name.c_str(), std::ios::binary);
+                std::vector<char> buf;
+
+                if (!file.good()) 
+                {
+                    throw runtime_error("Could not open file " + name);
+                }
+
+                file.unsetf(std::ios::skipws);
+                file.seekg(0, std::ios::end);
+                const size_t size = file.tellg();
+
+                file.seekg(0, std::ios::beg);
+                buf.resize(size);
+                file.read(buf.data(), size);
+                file.close();
+
+                return buf;
+            }
+
+            std::shared_ptr<unit_test::mock_snmp_client> mockSnmpClient;
+            std::shared_ptr<tmx::utils::MockUpdServer> mockUdpServer;
+
+            std::unique_ptr<SignalControllerConnection> signalControllerConnection;
+    };
+
+    TEST_F(TestSignalControllerConnection, initialize) {
+        tmx::utils::snmp_response_obj set_value;
+        set_value.type = tmx::utils::snmp_response_obj::response_type::INTEGER;
+        set_value.val_int = 2;
+        EXPECT_CALL(*mockSnmpClient, process_snmp_request("1.3.6.1.4.1.1206.3.5.2.9.44.1.0", tmx::utils::request_type::SET, set_value)).WillOnce(testing::DoAll(SetArgReferee<2>(set_value), Return(true)));
+        EXPECT_TRUE(signalControllerConnection->initializeSignalControllerConnection());
+    }
+
+    TEST_F(TestSignalControllerConnection, receiveBinarySPAT) {
+        auto spat_binary_buf = read_binary_file("../../SpatPlugin/test/test_spat_binaries/spat_1721238398773.bin");
```

Comment:
is it ok with this shorter path `./test_spat_binaries/spat_1721238398773.bin` if using relative path?

Reply:
That is not the correct relative path. This file is created under `src/v2i-hub/build/SpatPlugin/` and the file is under `src/v2i-hub/SpatPlugin/test`